specs/phase0/weak-subjectivity.md

```diff
@@ -120,7 +120,7 @@ The check can be implemented in the following way:
 ```python
 def is_within_weak_subjectivity_period(store: Store, ws_state: BeaconState, ws_checkpoint: Checkpoint) -> bool:
     # Clients may choose to validate the input state against the input Weak Subjectivity Checkpoint
-    assert ws_state.latest_block_header.state_root == ws_checkpoint.root
+    assert ws_state.latest_block_header.state_root == store.block_states[ws_checkpoint.root]
```

Comment:
This should be the `hash_tree_root`  of `store.block_states[ws_checkpoint.root]` right?

Reply:
Also, `latest_block_header.state_root` is often just `Bytes32()` because it avoids a circular dependency

You need to instead do something like the following:
```python
    latest_block_state_root = ws.latest_block_header.state_root
    if latest_block_state_root == Bytes32():
        latest_block_state_root = hash_tree_root(ws_state)
    assert latest_block_state_root == hash_tree_root(store.block_states[ws_checkpoint.root])
```

Reply:
> This should be the `hash_tree_root` of `store.block_states[ws_checkpoint.root]` right?

Yes. My bad.

```suggestion
    assert ws_state.latest_block_header.state_root == hash_tree_root(store.block_states[ws_checkpoint.root])
```

Reply:
> Also, latest_block_header.state_root is often just Bytes32() because it avoids a circular dependency

This breaks the functionality that we seek when bootstrapping from a WS checkpoint & associated state. We want to ensure that the given checkpoint & state are for the same point in the chain. 

A state's `latest_block_header.state_root` is `Bytes32()` if it's a state at the tip of the chain. For purposes of WS bootstrapping, only finalized states are expected to be used. I think it's ok to let states at the tip of the chain fail this method (it's unsafe to proceed anyway).

Reply:
finalized states also can have `Bytes32()` as the latest header in there. It's just post-states that are not skip slots and such a post state can easily be finalized

Reply:
> finalized states also can have Bytes32() as the latest header in there
Oops, looks like I overlooked this part. In that case, this function is broken & needs fixing.